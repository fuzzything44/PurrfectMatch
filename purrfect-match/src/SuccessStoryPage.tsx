import React from 'react';
import { Modal } from "./Modal";
import cat from "./pictures/cat.jpg";
import { Redirect } from "react-router-dom";

interface SuccessStoryPageProps {
};

interface SuccessStoryPageState {
    modalOpen: boolean;
};

export class SuccessStoryPage extends React.Component<SuccessStoryPageProps, SuccessStoryPageState> {

    constructor(props: any) {
        super(props);

        this.state = {
            modalOpen: false
        };
    }
    public render() {
        return (
            <div className="bgred">
                <Modal display={this.state.modalOpen} onClose={() => this.setState({ modalOpen: false })}>
                    <img id="successStoryPageImage" src={cat} alt="Cat"/> 
                    <div id="successStoryPageText">
                        <h1 id="successStoryPageHeader"> 
                            Name: Ollie
                            <br />Age: 3 years
                            <br />Breed: American Shorthair
                            <br />Adopted: 10/24/2019
                            <br />
                            <br />
                            A message from Ollie's owner, Bob:
                        </h1>
                        <p> 
                            Ollie has been a real friend! Coming home from a crazy day at work, I get to look forward to this goofball greeting me with lots of purring and cuddles. I found Ollie through Purrfect Match. I couldn’t go to my local shelter without first knowing if I had a specific pet in mind, because I can’t be away from work for very long during business hours. Perrfect Match helped me find Ollie, and I’m so happy he’s in my life! He’s a cutie, and I can’t imagine not having found him.
                        </p>
                    </div>
                </Modal>
                <div> Open popup: <button onClick={() => { this.setState({ modalOpen: true }) }}>button</button> </div>
            </div>);
    }

}


