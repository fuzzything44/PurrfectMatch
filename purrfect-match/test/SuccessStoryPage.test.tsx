import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { SuccessStoryPage } from 'SuccessStoryPage';

configure({ adapter: new Adapter() });

let wrapper: ShallowWrapper<any, any, any>;
beforeEach(() => {
    wrapper = shallow(<SuccessStoryPage />);
});

test("Snapshot test", () => {
    expect(wrapper).toMatchSnapshot();
});

test.todo("Gets stories from api call");
test.todo("Page change fetches new stories correctly");