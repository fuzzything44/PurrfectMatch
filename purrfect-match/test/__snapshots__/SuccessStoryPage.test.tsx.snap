// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div>
  <div
    className="welcomeMessage"
  >
    Did you adopt a pet using our site? Share your story here!
  </div>
  <div>
    <SubmissionBox />
  </div>
  <div
    className="successStoryHolder"
  >
    <div
      id="successStories"
    >
      <div
        id="headerOne"
      >
        Success Stories
      </div>
      <ul>
         
        <li>
           
          <CompactView
            info={
              Object {
                "adoptionDate": "10/24/19",
                "age": "4 years",
                "breed": "American shorthair",
                "gender": "Female",
                "image": "cat.jpg",
                "name": "Mittens",
                "owner": "Bob",
                "storyText": "Some story about how this cat was saved from horrible living conditions and is doing much better now.",
              }
            }
          />
           
        </li>
        <li>
           
          <CompactView
            info={
              Object {
                "adoptionDate": "10/24/19",
                "age": "2 months",
                "breed": "American shorthair",
                "gender": "Female",
                "image": "Cat_1.png",
                "name": "Fluffy",
                "owner": "Bob",
                "storyText": "Another success story here",
              }
            }
          />
           
        </li>
        <li>
           
          <CompactView
            info={
              Object {
                "adoptionDate": "10/24/19",
                "age": "10 years",
                "breed": "American shorthair",
                "gender": "Female",
                "image": "Dog_1.png",
                "name": "Mr. Woof",
                "owner": "Bob",
                "storyText": "Okay, success stories really should be longer but eh. ",
              }
            }
          />
           
        </li>
      </ul>
    </div>
    <div
      id="recentlyAdopted"
    >
      <div
        id="headerTwo"
      >
        Recently adopted!
      </div>
      <ul
        id="recentAdoptList"
      >
         
        <li>
           
          <CompactViewRecent
            info={
              Object {
                "adoptionDate": "10/24/19",
                "age": "4 years",
                "breed": "American shorthair",
                "gender": "Female",
                "image": "cat.jpg",
                "name": "Mittens",
                "owner": "Bob",
              }
            }
          />
           
        </li>
        <li>
           
          <CompactViewRecent
            info={
              Object {
                "adoptionDate": "10/24/19",
                "age": "2 months",
                "breed": "American shorthair",
                "gender": "Female",
                "image": "Cat_1.png",
                "name": "Fluffy",
                "owner": "Bob",
              }
            }
          />
           
        </li>
        <li>
           
          <CompactViewRecent
            info={
              Object {
                "adoptionDate": "10/24/19",
                "age": "10 years",
                "breed": "American shorthair",
                "gender": "Female",
                "image": "Dog_1.png",
                "name": "Mr. Woof",
                "owner": "Bob",
              }
            }
          />
           
        </li>
      </ul>
    </div>
  </div>
</div>
`;
